src/irmin-pack/s.ml

```diff
@@ -55,7 +55,34 @@ module type Specifics = sig
   (** [flush t] flush read-write pack on disk. Raises [RO_Not_Allowed] if called
       by a readonly instance.*)
 
-  val gc : ?unlink:bool -> repo -> commit_key -> unit
+  val start_gc :
+    ?unlink:bool ->
+    ?auto_finalise:bool ->
+    throttle:[ `Block | `Skip ] ->
+    repo ->
+    commit_key ->
+    bool Lwt.t
+  (** [start_gc] tries to start the gc process and returns true if the gc is
+      launched.
+
+      If [unlink] is false then temporary files on disk are kept after the gc
+      finished. This option is useful for debugging. The default is true.
+
+      If [auto_finalise] is true (the default) then there is no need to call
+      explicitly [finalise_gc].
+
+      If [throttle] is [Skip] and there is a concurrent gc, [start_gc] returns
+      false immediately without launching a second gc. If [throttle] is [Block]
+      and there is a concurrent gc, [start_gc] blocks waiting for the previous
+      gc to finish and only after launches a second gc. *)
```

Comment:
I would add:

If the previous GC failed, the function returns without launching a new GC.

TODO: Detail exceptions raised